module/dkg/broker.go

```diff
@@ -473,7 +480,7 @@ func (b *Broker) prepareBroadcastMessage(data []byte) (messages.BroadcastDKGMess
 //   or the signature could not be checked
 // TODO differentiate errors
 func (b *Broker) verifyBroadcastMessage(bcastMsg messages.BroadcastDKGMessage) (bool, error) {
-	err := b.checkMessageInstanceAndOrigin(bcastMsg.DKGMessage)
+	err := b.checkMessageInstanceAndOrigin(bcastMsg.Orig, bcastMsg.DKGMessage)
```

Comment:
- this function should not check the origin anymore, because we look up the origin from the local ID->index map. Since the local map is trusted by construction, the origin is also trusted. I suggest to rename it to `checkMessageInstance`. It is also called for incoming private messages, but the origin there shouldn't be checked either for the same reason.
- Let's update the doc too on line 474 and remove the origin mention.

Reply:
https://github.com/onflow/flow-go/pull/2238/commits/df9129e27e41825e65077a23fa471f3f37b239b2
https://github.com/onflow/flow-go/pull/2238/commits/152aca29e7879589906fabd7e14d52ea1d929cbb